hydrolib/core/dflowfm/tim/models.py

```diff
@@ -0,0 +1,73 @@
+from pathlib import Path
+from typing import Callable, Dict, List
+
+from pydantic.class_validators import validator
+
+from hydrolib.core.basemodel import ModelSaveSettings, ParsableFileModel
+
+from .parser import TimParser
+from .serializer import TimSerializer, TimSerializerConfig
+
+
+class TimModel(ParsableFileModel):
+    """Class representing a tim (*.tim) file.
+
+    Attributes:
+        data: Dictionary with keys \"comment\" & time as float and value as List of floats".\n
+        serializer_config: TimSerializerConfig
+
+    Raises:
+        ValueError: If the time and values are not parsable to float, or when the amount of columns differs per timeseries.
+    """
+
+    serializer_config = TimSerializerConfig()
+
+    comments: List[str]
+    timeseries: Dict[float, List[float]]
+
+    @classmethod
+    def _ext(cls) -> str:
+        return ".tim"
+
+    @classmethod
+    def _filename(cls) -> str:
+        return "sample"
+
+    @classmethod
+    def _get_serializer(
+        cls,
+    ) -> Callable[[Path, Dict, TimSerializerConfig, ModelSaveSettings], None]:
+        return TimSerializer.serialize
+
+    @classmethod
+    def _get_parser(cls) -> Callable:
+        return TimParser.parse
+
+    @validator("timeseries")
+    @classmethod
+    def _timeseries_values(
+        cls, v: Dict[float, List[float]]
+    ) -> Dict[float, List[float]]:
+        """Validates if the amount of columns per timeseries match.
+
+        Args:
+            v (Dict[float, List[float]]): Value to validate, the timeseries in this case.
+
+        Raises:
+            ValueError: If the time and values are not parsable to float, or when the amount of columns differs per timeseries.
```

Comment:
```suggestion
            ValueError: When the amount of columns differs per timeseries.
```

You do not need to specify the type validation, it is not the responsibility of this function.